core/src/worker/workflow/wft_poller.rs

```diff
@@ -119,7 +130,10 @@ impl WFTPollerShared {
                 } else {
                     let _ = non_sticky_active
                         .clone()
-                        .wait_for(|v| *v <= *sticky_active.borrow())
+                        .wait_for(|v| {
+                            *v < *sticky_active.borrow()
+                                || !self.have_done_first_poll.load(Ordering::Relaxed)
+                        })
```

Comment:
Is there an argument for a comment here to help future readers?

Reply:
For now, it looks to me like this condition will always eval to `true`, since `have_done_first_poll` is set to false on creation, and never changed afterward.

Reply:
Another question - Assuming `have_done_first_poll` would work as its name suggest, your change would prioritize sticky pollers over non-sticky except on the first polls. So assuming the case where max wft poller is one, it would forcibly be a sticky one from that point on, no? No more polling on the non-sticky tq, ever? Is that really the intended behavior?

Also, what if workflow cache is disabled (`max_cached_workflows = 0`), so that we don't even have a sticky task queue? I really don't know how this is handled internally, but it seems like everything would still go through `WFTPollerShared`, and so I assume that `sticky_active` would remain 0 forever in that case. And so we'd found ourselves in the situation of being incapable of doing anymore poll past the very first one.

Reply:
Yeah, you're right that those somehow just weren't used. They were before and then I refactored to simplify things and somehow this weird broken version actually does the right thing (probably accidentally by just not using sticky). Will fix.

Reply:
Added a comment as well